providers/azure_oidc.go

```diff
@@ -0,0 +1,112 @@
+package providers
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/pusher/oauth2_proxy/pkg/apis/sessions"
+	"github.com/pusher/oauth2_proxy/pkg/logger"
+	"golang.org/x/oauth2"
+)
+
+type AzureOIDCProvider struct {
+	*OIDCProvider
+	PermittedGroups []string
+	ExemptedEmails  []string
+}
+
+type AzureOIDCClaims struct {
+	*OIDCClaims
+	Groups []string
+}
+
+// NewAzureOIDCProvider initiates a new AzureOIDCProvider
+func NewAzureOIDCProvider(p *ProviderData) *AzureOIDCProvider {
+	provider := AzureOIDCProvider{
+		OIDCProvider: NewOIDCProvider(p),
+	}
+	provider.ProviderName = "Azure OpenID Connect"
+	provider.ApprovalPrompt = "consent"
```

Comment:
Can you make the ApprovalPrompt come from the input param instead of hardcoding as "consent"